### DISCUSSION TITLE
types of parameters

### DISCUSSION THREAD

**Discussion by MandoFi** [MilesCranmer/PySR#333] [General]
I am currently attempting to compare predictions of clinical outcomes (dead vs alive (logistic) or number of hospitalizations days(linear)) using different paramters (age (continuous), sex (categorica), bmi(continuous)) using traditional (logistic or linear) regression to symbolic regression. I was wondering how I can ensure that the "PySRRegressor" function to develop the different equations takes into account the different types variables/predictors (continuous or categorical).

**Comment by MilesCranmer** [comment]
Would you be able to clarify what you mean by "takes into account"? What kind of specialization would be used?

**Reply by MandoFi** [reply]
Dear Miles Cramer,

Thank you for your quick reply. I should have been clearer. How can I use PySR for classification/binary outcomes (dead or alive)? I was also wondering how I can tell the "PySRRegressor" which predictors/variables are continuous (BMI, age, etc...) and which are categorical (sex, hospital, history of heart disease, etc...). I've encoded categorical variables with numbers (0=female, male=1). I have a sense that the PySRRegressor function therefore increases the importance of continuous variables (BMI/age) disproportionately because there is a much wide range in BMI's (18 to 50) or age (18 to 95) than sex (being only 1=male or 0=female) or history of heart disease (yes=1 or no=0).

**Reply by MilesCranmer** [reply]
Unfortunately there is no way to make it treat features differently when they are discrete vs continuous. However, you can map discrete variables onto continuous quantities, and things usually work just fine. You can also use rounding operations in custom operators, like:
```
binary_operators=["mypow(x, y) = x ^ ceil(y)"]
```
would round `y` to the next largest integer.
